goal_map.py

```diff
@@ -1,12 +1,63 @@
-from helper_types import GoalState, GoalRegionObservation
+from helper_types import (
+    GoalRegionState,
+    GoalRegionObservation,
+    GoalRegion,
+    ExpectedGamePiece,
+)
+from wpimath.geometry import Translation3d
 
 
-class GoalMap:
-    def __init__(self):
-        pass
+class GoalRegionMap:
+    X_OFFSET_TO_GRID: float = 0.0
+    Y_OFFSET_TO_GRID: float = 516.763
+    INTAKE_ZONE_OFFSET: float = 0.0
+    Y_DISTANCE_BETWEEN_GOALS: float = 558.0
+
+    Z_DISTANCE_CUBE_LOOK_UP: list[float] = [1169.988, 865.188, 0.0]
+    Z_DISTANCE_CONE_LOOK_UP: list[float] = [
+        826.326 + 120.0,
+        522.288 + 120.0,
+        0.0 + 120.0,
+    ]
+
+    X_DISTANCE_CUBE_LOOK_UP: list[float] = [353.22, 796.777, 1197.227]
+    X_DISTANCE_CONE_LOOK_UP: list[float] = [364.231, 795.231, 1167.655]
+
+    def __init__(self, on_blue_alliance: bool):
+        self.map: list[GoalRegionState] = []
```

Comment:
This might be easier to understand if this were an ndarray - the grid is a fixed size after all.